libcnb/src/target.rs

```diff
@@ -1,35 +1,40 @@
+#[derive(Debug)]
 pub struct Target {
     /// The name of the target operating system.
     ///
     /// The value should conform to [Go's `$GOOS`](https://golang.org/doc/install/source#environment), for example
     /// `linux` or `windows`.
     ///
-    /// CNB `lifecycle` sources this value from the build OCI image's [`os` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    /// CNB `lifecycle` sources this value from the run OCI image's [`os` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
     pub os: String,
     /// The name of the target CPU architecture.
     ///
     /// The value should conform to [Go's $GOARCH](https://golang.org/doc/install/source#environment), for example
     /// `amd64` or `arm64`.
     ///
-    /// CNB `lifecycle` sources this value from the build OCI image's [`architecture` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    /// CNB `lifecycle` sources this value from the run OCI image's [`architecture` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
     pub arch: String,
     /// The variant of the specified CPU architecture.
     ///
     /// The value should conform to [OCI image spec platform variants](https://github.com/opencontainers/image-spec/blob/main/image-index.md#platform-variants), for example
     /// `v7` or `v8`.
     ///
-    /// CNB `lifecycle` sources this value from the build OCI image's [`variant` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
+    /// CNB `lifecycle` sources this value from the run OCI image's [`variant` property](https://github.com/opencontainers/image-spec/blob/main/config.md#properties).
     pub arch_variant: Option<String>,
     /// The name of the operating system distribution. Should be empty for Windows.
     ///
     /// For example: `ubuntu` or `alpine`.
     ///
-    /// CNB `lifecycle` sources this value from the build OCI image's `io.buildpacks.base.distro.name` label.
-    pub distro_name: Option<String>,
+    /// CNB `lifecycle` sources this value from either:
+    /// 1. The `io.buildpacks.base.distro.name` OCI image label, if set on the run image.
+    /// 2. Or else, the `ID` field of the `/etc/os-release` file in the build image.
```

Comment:
The labels are sourced from the run image, per:
https://github.com/buildpacks/spec/blob/platform/0.12/platform.md#target-data

However, the `/etc/os-release` fallback run against the build image, per:
https://github.com/buildpacks/lifecycle/blob/13d3691dd017405e22ad7b5050204cc2ccdf034a/platform/target_data.go#L16